### ISSUE TITLE
dev-lang/python sanbox violation with test_compileall

### ISSUE THREAD

**Issue by grigi** [InBetweenNames/gentooLTO#87]
Hi, when using the python 3.5 & 3.6 ebuilds with the `pgo` USE flag from your overlay I get a sandbox violation with the test `test_compileall`, when I remove it, it completes the build.

Another oddity is that `test_curses` leaves my terminal in an unusable state.

**Comment by InBetweenNames** [comment]
Thanks for the report!  Would you be willing to submit a PR to remove it from the test list?

**Comment by pchome** [comment]
https://github.com/InBetweenNames/gentooLTO/issues/20#issuecomment-331450256

But this days I able to build pgo-ed python 2.7 and 3.6 w/o this issue somehow. I don't even know what was changed.

Maybe python 3.5 removal or switching system's default version to 3.6 helped.
`$ eselect python list `
```sh
Available Python interpreters, in order of preference:
  [1]   python3.6
  [2]   python2.7
  [3]   python3.4 (uninstalled)
  [4]   python3.5 (uninstalled)
```

`$ grep PYTHON /etc/portage/make.conf`
```sh
PYTHON_TARGETS="python3_6 python2_7"
PYTHON_SINGLE_TARGET="python3_6"
```

**Comment by InBetweenNames** [comment]
Indeed, mine is set to Python 3.6 as well, and I also removed older Pythons from my system where possible.  @grigi , could you tell us what your interpreter is?

**Comment by grigi** [comment]
```eselect python list
Available Python interpreters, in order of preference:
  [1]   python2.7
  [2]   python3.5
  [3]   python3.4
  [4]   python3.6 (fallback)
  [5]   pypy3 (fallback)
  [6]   pypy (fallback)
```
But I don't have python3.4 installed anymore.

Hmm, why is py2.7 my default?

**Comment by InBetweenNames** [comment]
Can you try with Python 3.6 selected?

**Comment by grigi** [comment]
Hmm, I'm using the default USE flags on portage, which doesn't include the py3.6 libraries. I'll try with 3.5 and see if the issue is using py2.7.

**Comment by InBetweenNames** [comment]
@grigi Any word on if Python 3.5 helped?

**Comment by grigi** [comment]
Hi @InBetweenNames When changing my default python interpreter to 3.5 `test_compileall` still has a sandbox violation.

**Comment by grigi** [comment]
The diff to get 3.5 & 3.6 to build with `pgo` set for me:
```diff --git a/dev-lang/python/python-3.5.4-r2.ebuild b/dev-lang/python/python-3.5.4-r2.ebuild
index d5e1097..a57303d 100644
--- a/dev-lang/python/python-3.5.4-r2.ebuild
+++ b/dev-lang/python/python-3.5.4-r2.ebuild
@@ -176,7 +176,7 @@ src_compile() {
 
        cd "${BUILD_DIR}" || die
        if use pgo; then
-               emake profile-opt PROFILE_TASK="-m test.regrtest -w -uall,-audio -x test_gdb test_multiprocessing test_subprocess test_tokenize test_signal test_faulthandler test_asyncio"
+               emake profile-opt PROFILE_TASK="-m test.regrtest -w -uall,-audio -x test_gdb test_multiprocessing test_subprocess test_tokenize test_signal test_faulthandler test_asyncio test_curses test_compileall"
        else
                emake CPPFLAGS= CFLAGS= LDFLAGS=
        fi
diff --git a/dev-lang/python/python-3.6.3-r2.ebuild b/dev-lang/python/python-3.6.3-r2.ebuild
index d1a408b..e3ae8fe 100644
--- a/dev-lang/python/python-3.6.3-r2.ebuild
+++ b/dev-lang/python/python-3.6.3-r2.ebuild
@@ -161,7 +161,7 @@ src_compile() {
        local -x LC_ALL=C
 
        if use pgo; then
-               emake profile-opt PROFILE_TASK="-m test.regrtest -w -uall,-audio -x test_gdb test_multiprocessing test_subprocess test_tokenize test_signal test_faulthandler test_asyncio"
+               emake profile-opt PROFILE_TASK="-m test.regrtest -w -uall,-audio -x test_gdb test_multiprocessing test_subprocess test_tokenize test_signal test_faulthandler test_asyncio test_compileall test_curses"
        else
                emake CPPFLAGS= CFLAGS= LDFLAGS=
        fi
```
As I originally said, disabling `test_curses` isn't because of a sandbox violation, but because it breaks the console.

**Comment by InBetweenNames** [comment]
Sorry for the delay in getting back to you.  Thanks for the detailed followup -- I will accept a PR to amend the 3.6.4 ebuild to correct this.  I see no harm in removing the two tests as we still have 398 other intensive tests to get the PGO information from.  Users who are interested in training using all tests can edit the ebuild to do so.

**Comment by InBetweenNames** [comment]
Should be fixed in HEAD ( #95 )
